NoSuchMethodError when formatting with JDK 21

## What happened?

When compiling maven master branch using JDK 21, I have the following exception:
```
Caused by: java.lang.NoSuchMethodError: 'com.sun.tools.javac.tree.JCTree com.sun.tools.javac.tree.JCTree$JCImport.getQualifiedIdentifier()'
	at com.palantir.javaformat.java.RemoveUnusedImports.getSimpleName(RemoveUnusedImports.java:245)
	at com.palantir.javaformat.java.RemoveUnusedImports.buildReplacements(RemoveUnusedImports.java:225)
	at com.palantir.javaformat.java.RemoveUnusedImports.removeUnusedImports(RemoveUnusedImports.java:209)
	at com.diffplug.spotless.glue.pjf.PalantirJavaFormatFormatterFunc.apply(PalantirJavaFormatFormatterFunc.java:42)
```

## What did you want to happen?

The formatting should work !

## Analysis

The [`getSimpleName` method](https://github.com/palantir/palantir-java-format/blob/develop/palantir-java-format/src/main/java/com/palantir/javaformat/java/RemoveUnusedImports.java#L244-L250) needs to adapted somehow.  The JDK internal class `JCImport` [has changed recently](https://github.com/openjdk/jdk/commit/a917fb3fcf0fe1a4c4de86c08ae4041462848b82#diff-1fa76229dfd36a4b7f96a49c14157c3eac397346221b159f82c9963df8903fb5R682).  In [JDK 17](https://github.com/openjdk/jdk/blob/jdk-17%2B35/src/jdk.compiler/share/classes/com/sun/tools/javac/tree/JCTree.java#L677), the method was defined as:
```
public JCTree getQualifiedIdentifier() { return qualid; }
```
and in [JDK 21](https://github.com/openjdk/jdk/blob/jdk-21%2B27/src/jdk.compiler/share/classes/com/sun/tools/javac/tree/JCTree.java#L686):
```
public JCFieldAccess getQualifiedIdentifier() { return qualid; }
```
One possibility to fix that would be to fix the `getSimpleName()` method to use an argument of type `com.sun.source.tree.ImportTree` and call the `getQualifiedIdentifier()` method which returns a `com.sun.source.tree.Tree` and has not changed.

```
    private static String getSimpleName(ImportTree importTree) {
        return importTree.getQualifiedIdentifier() instanceof JCIdent
                ? ((JCIdent) importTree.getQualifiedIdentifier()).getName().toString()
                : ((JCFieldAccess) importTree.getQualifiedIdentifier())
                        .getIdentifier()
                        .toString();
    }
```

--- a/palantir-java-format/src/main/java/com/palantir/javaformat/java/RemoveUnusedImports.java
+++ b/palantir-java-format/src/main/java/com/palantir/javaformat/java/RemoveUnusedImports.java
@@ -241,7 +241,7 @@ public class RemoveUnusedImports {
         return replacements;
     }
 
-    private static String getSimpleName(JCImport importTree) {
+    private static String getSimpleName(ImportTree importTree) {
         return importTree.getQualifiedIdentifier() instanceof JCIdent
                 ? ((JCIdent) importTree.getQualifiedIdentifier()).getName().toString()
                 : ((JCFieldAccess) importTree.getQualifiedIdentifier())

